public/Connect-DbaInstance.ps1

```diff
@@ -645,15 +645,39 @@ function Connect-DbaInstance {
             } elseif ($inputObjectType -eq 'SqlConnection') {
                 $server = New-Object -TypeName Microsoft.SqlServer.Management.Smo.Server -ArgumentList $inputObject
             } elseif ($inputObjectType -in 'RegisteredServer', 'ConnectionString') {
-                $server = New-Object -TypeName Microsoft.SqlServer.Management.Smo.Server -ArgumentList $serverName
-                # Parameter TrustServerCertificate changes the connection string be allow connections to instances with the default self-signed certificate
+                # Create the server SMO in the same way as when passing a string (see #8962 for details).
+                # Best way to get connection pooling to work is to use SqlConnectionInfo -> ServerConnection -> Server
+                $sqlConnectionInfo = New-Object -TypeName Microsoft.SqlServer.Management.Common.SqlConnectionInfo
```

Comment:
Not required now but since we formally only support PowerShell v5+ now you can use this and it adds some speed to the commands:

```suggestion
                $sqlConnectionInfo = [Microsoft.SqlServer.Management.Common.SqlConnectionInfo]::new()
```

Reply:
I love the new syntax a lot and use it in all of m own code. But I think we still use the old syntax for compatibility.

Reply:
Ah, sorry updated comment this syntax started working PS v3. Since we have formally dropped support for PS2 we can use this now.

Reply:
But then I would like to change this all other the command or even all over the code. And I would like to change that in a separate PR. Ok?

Reply:
Noooooooooo, you forever every couple years, Shawn 😅  We cannot support ::new

Reply:
Wheover said it works in PSv3 is mistaken. I tried it on Windows 7 with PS 3 and it does not work